uber/models.py

```diff
@@ -1142,8 +1127,13 @@ def _badge_adjustments(self):
             if self.orig_value_of('badge_num'):
                 self.session.shift_badges(self.orig_value_of('badge_type'), self.orig_value_of('badge_num') + 1, down=True)
             self.badge_num = None
-        elif self.session.needs_badge_num(self):
-            if not self.badge_num:
+        elif self.session.needs_badge_num(self) and not self.badge_num:
+            self.badge_num = self.session.get_next_badge_num(self.badge_type)
+
+        if self.badge_num:
```

Comment:
Goldangit why do we have duplicate numbers >:[

This block of code is another, deeper-level guard against duplicate numbers.

It also means that we can run presave_adjustments on all the badges to fix our current numbers. Yay!